src/easylink/rule.py

```diff
@@ -64,37 +81,40 @@ def _build_rule(self) -> str:
 
 @dataclass
 class ImplementedRule(Rule):
-    """A rule that defines the execution of an implementation"""
+    """A :class:`Rule` that defines the execution of an :class:`~easylink.implementation.Implementation`."""
 
     name: str
-    """Name to give the rule."""
+    """Name of the rule."""
     step_name: str
-    """Name of the step."""
+    """Name of the step this rule/``Implementation`` is implementing."""
     implementation_name: str
-    """Name of the implementation."""
+    """Name of the ``Implementation`` to build the rule for."""
     input_slots: dict[str, dict[str, str | list[str]]]
-    """List of file paths required by implementation."""
+    """This ``Implementation's`` input slot attributes."""
```

Comment:
Why isn't this just on the Implementation?

Reply:
It is, and @patricktnast and I talked about [this sort of pattern](https://github.com/ihmeuw/easylink/blob/main/src/easylink/pipeline.py#L95) we've been using quite a bit yesterday. I think the goal was to not keep sending entire classes around too much? Though with embarrassingly parallel stuff I'm looking at doing a very similar thing.

There have been quite a few discussions about possible refactors in the future for sure.

Reply:
Gotcha. Personally I don't think sending entire classes around is a big problem, though I haven't thought much about it. Sounds like you all are way ahead of me here, no need to do anything right now 😄